geopandas/base.py

```diff
@@ -27,13 +27,20 @@ def is_geometry_type(data):
 
 def _delegate_binary_method(op, this, other, align, *args, **kwargs):
     # type: (str, GeoSeries, GeoSeries) -> GeoSeries/Series
+    if align is None:
+        align = True
+        maybe_warn = True
+    else:
+        maybe_warn = False
     this = this.geometry
     if isinstance(other, GeoPandasBase):
         if align and not this.index.equals(other.index):
-            warn(
-                "The indices of the two GeoSeries are different.",
-                stacklevel=4,
-            )
+            if maybe_warn:
+                warn(
+                    "The indices of the two GeoSeries are different and will be "
+                    "aligned. Use the `align` keyword to control this behavior.",
```

Comment:
Do we want the expand the warning message as suggested in the issue?

Reply:
Forgot about that. Now in 84fff16